src/Evolution/Particles/MonteCarlo/EvolveMonteCarloPackets.hpp

```diff
@@ -0,0 +1,61 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <vector>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "NumericalAlgorithms/Spectral/Mesh.hpp"
+#include "Utilities/Gsl.hpp"
+
+/// Items related to the evolution of particles
+namespace Particles {
+
+/// Items related to Monte-Carlo radiation transport
+namespace MonteCarlo {
+
+struct Packet;
+
+namespace detail {
+
+// Time derivative of the spatial component of the momentum one-form on a null
+// geodesic
+void time_derivative_momentum_geodesic(
+    const gsl::not_null<std::vector<double>*> dt_momentum, const Packet& packet,
```

Comment:
When you rebase, clang-tidy may ask you to change this to just `gsl::not_null...` as in, remove the const. Same on lines 35 and 49below.